cabal-install/Distribution/Client/SetupWrapper.hs

```diff
@@ -334,7 +335,8 @@ selfExecSetupMethod verbosity options _pkg bt mkargs = do
 
   searchpath <- programSearchPathAsPATHVar
                 (getProgramSearchPath (useProgramConfig options))
-  env        <- getEffectiveEnvironment [("PATH", Just searchpath)]
```

Comment:
FWIW, his idea would make sense if we wanted to propagate `HASKELL_DIST_DIR` automatically to any sub-cabal process—*not* what this PR is about.
In scenarios that do need that for whatever reason (assuming they exist, and maybe they do), scripts can/should use `--builddir=$HASKELL_DIST_DIR`.